lib/scope.js

```diff
@@ -72,6 +72,9 @@ SymbolDef.prototype = {
             || this.orig[0] instanceof AST_SymbolMethod
             || (options.keep_classnames
                 && (this.orig[0] instanceof AST_SymbolClass
+                    || this.orig[0] instanceof AST_SymbolDefClass))
+            || (options.reserved && options.reserved.indexOf(this.name) >= 0
+                && (this.orig[0] instanceof AST_SymbolClass
```

Comment:
We don't filter `reserved` variable names here. It is done inside [`mangle_names()`](https://github.com/mishoo/UglifyJS2/blob/e0e009ace25cc6fdaf99d103a8c680ad670d1eed/lib/scope.js#L548) instead.